### PR TITLE
Issue 3952 improve text clarity with password protected email

### LINKED ISSUE #3952: improve text sent with password protected email
Currently it says:

> This message is encrypted [Open Message]
> 
> Alternatively copy and paste the following link: https://...

Let's give the recipient more context, as follows:

> sender@example.com has sent you a password-encrypted email. [click here to Open Message]
> 
> Alternatively copy and paste the following link: https://...

### PR DESCRIPTION
**Pull request by martgil**
This PR will  improve text clarity with password-protected email

close #3952 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)
- Tests will be added later (issue #...)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Review by rrrooommmaaa — CHANGES_REQUESTED** [review]

File: extension/js/common/lang.ts [hunk 1]
```diff
@@ -66,10 +66,10 @@ export const Lang = { // tslint:disable-line:variable-name
   compose: {
     abortSending: 'A message is currently being sent. Closing the compose window may abort sending the message.\nAbort sending?',
     pleaseReconnectAccount: 'Please reconnect FlowCrypt to your Gmail Account. This is typically needed after a long time of no use, a password change, or similar account changes. ',
-    msgEncryptedHtml: { EN: 'This&nbsp;message&nbsp;is&nbsp;encrypted: ', DE: 'Diese&nbsp;Nachricht&nbsp;ist&nbsp;verschlüsselt: ' },
-    msgEncryptedText: { EN: 'This message is encrypted. Follow this link to open it: ', DE: 'Diese Nachricht ist verschlüsselt. Nachricht öffnen: ' },
+    msgEncryptedHtml: (lang: string, acctEmail: string) =>  lang === 'EN' ? `${acctEmail}&nbsp;has&nbsp;sent&nbsp;you&nbsp;a&nbsp;password-encrypted&nbsp;email ` : `${acctEmail}&nbsp;hat&nbsp;Ihnen&nbsp;eine&nbsp;passwortverschlüsselte&nbsp;E-Mail&nbsp;gesendet `,
+    msgEncryptedText: (lang: string, acctEmail: string) => lang === 'EN' ? `${acctEmail} has sent you a password-encrypted email. Follow this link to open it: ` : `${acctEmail} hat Ihnen eine passwortverschlüsselte E-Mail gesendet. Folgen Sie diesem Link, um es zu öffnen `,
```
[the remark this answers is not in the document]
Comment (martgil) [reply extension/js/common/lang.ts:70]: Oh, yes! It should definitely work like that! I'll be working on it. Thank you sir!

File: extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts [hunk 2]
```diff
@@ -231,8 +231,8 @@ export class EncryptedMsgMailFormatter extends BaseMailFormatter {
       text.push(intro + '\n');
       html.push(Xss.escape(intro).replace(/\n/g, '<br>') + '<br><br>');
     }
-    text.push(Lang.compose.msgEncryptedText[lang] + msgUrl + '\n\n');
-    html.push(`${Lang.compose.msgEncryptedHtml[lang] + a}<br/><br/>${Lang.compose.alternativelyCopyPaste[lang] + Xss.escape(msgUrl)}<br/><br/>`);
+    text.push(Lang.compose.msgEncryptedText(lang, Xss.escape(this.acctEmail)) + msgUrl + '\n\n');
```
[the remark this answers is not in the document]
Comment (martgil) [reply extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts:234]: Copy sir, @rrrooommmaaa. Thanks for the very helpful context here. I'll be working on this as soon as possible.

**Comment by martgil** [comment]
sir, @rrrooommmaaa. I was able to follow your given instructions which found helpful as usual. When I tried to get combine it and try to read the sent email by email subject using 
```javascript
      const sentMsg = (await GoogleData.withInitializedData('flowcrypt.compatibility@gmail.com')).getMessageBySubject(subject)!;
```
From here I'm getting undefined as the returned value.  I found out that the function `getMessageBySubject` pulls some data from `AcctDataFile`? Is there any chance that I need to create another file in the project which contains a message and subject that matches the same subject that I'm including in this test? The specific subject that I'm using is 
```javascript
      const subject = 'PWD encrypted message with flowcrypt.com/api';
```
which was later could be use in `TestBySubjectStrategyContext`

Thanks in advance.

**Comment by rrrooommmaaa** [comment]
Hi @martgil I think we don't need to extract the sent message like this.
Messages with a subject containing 'PWD encrypted message with flowcrypt.com/api' will go to `PwdEncryptedMessageWithFlowCryptComApiTestStrategy`. It doesn't save the message to DATA, like some other "strategies" do, like `SaveMessageInStorageStrategy`.
Other strategies just make some checks and possibly throw an exception, thus making the test fail.
It's enough to set breakpoint in the PwdEncryptedMessageWithFlowCryptComApiTestStrategy, make sure it is actually hit (I've already added a check to compare text with sender e-mail address there -- you should see if you pulled my update) and see if any exception is thrown there in all 3 tests. This is what I meant by 'verification happens in some other place' -- that is, we don't need to verify in the test body. We can verify in this strategy -- it has all the necessary input -- incoming e-mail with "From" and the text. We can simply leave a comment in the test body like Tom did:
```
      // this test is using PwdEncryptedMessageWithFlowCryptComApiTestStrategy to check sent result based on subject "PWD encrypted message with flowcrypt.com/api"
```

**Comment by martgil** [comment]
Ah, you're right @rrrooommmaaa. I finally get it now :) Thank you so much!!

**Review by rrrooommmaaa — APPROVED** [review]

**Pull request merged**